src/ILCompiler.Compiler/src/JitInterface/TypesDebugInfo/UserDefinedTypeDescriptor.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+using Internal.TypeSystem;
+using ILCompiler.DependencyAnalysis;
+using System.Reflection.Metadata;
+using Internal.TypeSystem.Ecma;
+
+namespace Internal.JitInterface.TypesDebugInfo.UserDefinedTypes
+{
+
+    internal struct EnumRecordTypeDescriptor
+    {
+        public ulong Value;
+        public string Name;
+    };
+
+    internal struct EnumTypeDescriptor
+    {
+        public ulong ElementType;
+        public ulong ElementCount;
+        public string Name;
+        public string UniqueName;
+    }
+
+
+    internal class UserDefinedTypeDescriptor
+    {
+        public UserDefinedTypeDescriptor(ObjectWriter objectWriter)
```

Comment:
This kind of looks like a layering violation. A file under `JitInterface` shouldn't really depend on the `ObjectWriter`. We had a different object writer in the past (one that produced `.S` assembly files). Maybe we will write another object writer that will be fully in C# in the future, etc.

Having this class here (and using the structs it defines as direct ABI with the object writer itself - where you pass them to native functions) seems pretty inflexible.